packages/block-library/src/navigation/test/use-navigation-menu.js

```diff
@@ -0,0 +1,236 @@
+	const dispatch = registry.dispatch( coreStore );
+	dispatch.receiveUserPermission( 'create/navigation', allowed );
+	dispatch.startResolution( 'canUser', [ 'create', 'navigation' ] );
+	dispatch.finishResolution( 'canUser', [ 'create', 'navigation' ] );
+}
+
+function resolveUpdatePermission( registry, ref, allowed ) {
+	const dispatch = registry.dispatch( coreStore );
+	dispatch.receiveUserPermission( `update/navigation/${ ref }`, allowed );
+	dispatch.startResolution( 'canUser', [ 'update', 'navigation', ref ] );
+	dispatch.finishResolution( 'canUser', [ 'update', 'navigation', ref ] );
+}
+
+function resolveDeletePermission( registry, ref, allowed ) {
+	const dispatch = registry.dispatch( coreStore );
+	dispatch.receiveUserPermission( `delete/navigation/${ ref }`, allowed );
+	dispatch.startResolution( 'canUser', [ 'delete', 'navigation', ref ] );
+	dispatch.finishResolution( 'canUser', [ 'delete', 'navigation', ref ] );
+}
+
+describe( 'useNavigationMenus', () => {
+	const navigationMenu1 = { id: 1, title: 'Menu 1', status: 'publish' };
+	const navigationMenu2 = { id: 2, title: 'Menu 2', status: 'publish' };
+	const navigationMenu3 = { id: 3, title: 'Menu 3', status: 'publish' };
+	const navigationMenus = [
+		navigationMenu1,
+		navigationMenu2,
+		navigationMenu3,
+	];
+
+	let registry;
+	beforeEach( () => {
+		registry = createRegistryWithStores();
+		useSelect.mockImplementation( ( fn ) => fn( registry.select ) );
+	} );
+
+	it( 'Should return no information when no data is resolved', () => {
+		expect( useNavigationMenu() ).toEqual( {
+			navigationMenus: null,
+			canSwitchNavigationMenu: false,
+			canUserCreateNavigationMenu: false,
+			canUserDeleteNavigationMenu: false,
+			canUserUpdateNavigationMenu: false,
+			hasResolvedCanUserCreateNavigationMenu: false,
+			hasResolvedCanUserDeleteNavigationMenu: false,
+			hasResolvedCanUserUpdateNavigationMenu: false,
+			hasResolvedNavigationMenus: false,
+			isNavigationMenuMissing: true,
+			isNavigationMenuResolved: false,
+			isResolvingCanUserCreateNavigationMenu: false,
+			isResolvingNavigationMenus: false,
+		} );
+	} );
+
+	it( 'Should return information about all menus when ref is missing', () => {
+		resolveRecords( registry, navigationMenus );
+		resolveCreatePermission( registry, true );
+		expect( useNavigationMenu() ).toEqual( {
+			navigationMenus,
+			canSwitchNavigationMenu: true,
+			canUserCreateNavigationMenu: true,
+			canUserDeleteNavigationMenu: false,
+			canUserUpdateNavigationMenu: false,
+			hasResolvedCanUserCreateNavigationMenu: true,
+			hasResolvedCanUserDeleteNavigationMenu: false,
+			hasResolvedCanUserUpdateNavigationMenu: false,
+			hasResolvedNavigationMenus: true,
+			isNavigationMenuMissing: true,
+			isNavigationMenuResolved: false,
+			isResolvingCanUserCreateNavigationMenu: false,
+			isResolvingNavigationMenus: false,
+		} );
+	} );
+
+	it( 'Should also return information about a specific menu when ref is given', () => {
```

Comment:
```suggestion
	it( 'Should return information about a specific menu when ref is given', () => {
```

Nit: I think we can drop the "also" here